pkgs/applications/window-managers/phosh/default.nix

```diff
@@ -0,0 +1,150 @@
+{ stdenv
+, pkgs
+, fetchFromGitLab
+, substituteAll
+, meson
+, ninja
+, pkgconfig
+, wrapGAppsHook
+, libhandy
+, pulseaudio
+, glib
+, gtk3
+, gnome3
+, gcr
+, pam
+, upower
+, wayland
+, dbus
+, xvfb_run
+, phoc
+, feedbackd
+, squeekboard
+, networkmanager
+, polkit
+, elogind
+, libsecret
+, git
+, writeText
+, makeWrapper
+, makeDesktopItem
+}:
+
+let
+  gvc = fetchFromGitLab {
+    domain = "gitlab.gnome.org";
+    owner = "GNOME";
+    repo = "libgnome-volume-control";
+    rev = "ae1a34aafce7026b8c0f65a43c9192d756fe1057";
+    sha256 = "0a4qh5pgyjki904qf7qmvqz2ksxb0p8xhgl2aixfbhixn0pw6saw";
+  };
+
+  # The upstream desktop file in phosh repo points to a stub executable
+  oskDesktop = makeDesktopItem {
+    name = "sm.puri.OSK0";
+    type = "Application";
+    desktopName = "On-screen keyboard";
+    exec = "${squeekboard}/bin/squeekboard";
+    categories = "GNOME;Core;";
+    extraEntries = ''
+      OnlyShowIn=GNOME;
+      NoDisplay=true
+      X-GNOME-Autostart-Phase=Panel
+      X-GNOME-Provides=inputmethod
+      X-GNOME-Autostart-Notify=true
+      X-GNOME-AutoRestart=true
+    '';
+  };
+
+in stdenv.mkDerivation rec {
+  pname = "phosh";
+  version = "0.3.0";
+
+  src = fetchFromGitLab {
+    domain = "source.puri.sm";
+    owner = "Librem5";
+    repo = pname;
+    rev = "v${version}";
```

Comment:
I can't fetch this tag containing a `+` sign - it might be our `fetchFromGitLab` fetcher isn't escaping, where it should:
```
trying https://source.puri.sm/api/v4/projects/Librem5%2Ffeedbackd/repository/archive.tar.gz?sha=v0.0.0+git20200420
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
curl: (22) The requested URL returned error: 404
error: cannot download source from any mirror
```

(you might need to set the `sha256` to all-zeroes to reproduce this)

Reply:
Oh, I see. Should I use again the commit hash or I duplicate the `rev` value with `v0.0.0%2Bgit20200420` ?

Reply:
You should be able to use `"v${stdenv.lib.replaceStrings [ "+" ] [ "%2B" ] version}"`.

Reply:
I fixed the underlying issue in https://github.com/NixOS/nixpkgs/pull/88810.

Reply:
done